cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternalsCommon.qll

```diff
@@ -188,6 +188,93 @@ private class PointerOrReferenceTypeIndirection extends Indirection instanceof P
   override predicate isAdditionalWrite(Node0Impl value, Operand address, boolean certain) { none() }
 }
 
+private module IteratorIndirections {
+  import semmle.code.cpp.models.interfaces.Iterator as Interfaces
+  import semmle.code.cpp.models.implementations.Iterator as Iterator
+  import semmle.code.cpp.models.implementations.StdContainer as StdContainer
+
+  class IteratorIndirection extends Indirection instanceof Interfaces::Iterator {
+    IteratorIndirection() {
+      not this instanceof PointerOrReferenceTypeIndirection and baseType = super.getValueType()
+    }
+
+    override int getNumberOfIndirections() { result = 1 + countIndirections(this.getBaseType()) }
```

Comment:
Since this is not a pointer or a reference, can this be anything else than `1`?

Reply:
If `it` is an iterator that wraps an `int*`, then this call to `getNumberOfIndirections` will be 1 plus the number of indirections for `int*`. So I _think_ this is correct 🤔

Reply:
Does base type here refer to the type being wrapped by the iterator?

Reply:
Yep. [getBaseType](https://github.com/github/codeql/blob/mathiasvp/replace-ast-with-ir-use-usedataflow/cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternalsCommon.qll#L169) returns the unspecified version of the type returned by `super.getValueType()` (which I added to iterators in https://github.com/github/codeql/pull/11537).

Reply:
Cool. Apologies for the naive questions. I don't have all the context here.

Reply:
No need to apologize 😄!